### ISSUE TITLE
0.23.0a1 version causing INTERNALERROR at startup

### ISSUE THREAD

**Issue by blink1073** [pytest-dev/pytest-asyncio#679]
We run our `ipykernel` tests against prereleases in addition to regular tests, and it is failing with this error:

```
 ============================= test session starts ==============================
platform linux -- Python 3.11.6, pytest-7.4.3, pluggy-1.3.0 -- /home/runner/.local/share/hatch/env/virtual/ipykernel/_P2Rok5v/test/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/ipykernel/ipykernel
configfile: pyproject.toml
testpaths: tests, tests/inprocess
plugins: timeout-2.2.0, cov-4.1.0, flaky-3.7.0, asyncio-0.23.0a1
...
INTERNALERROR>   File "/home/runner/.local/share/hatch/env/virtual/ipykernel/_P2Rok5v/test/lib/python3.11/site-packages/pytest_asyncio/plugin.py", line 577, in pytest_collectstart
INTERNALERROR>     scope=_fixture_scope_by_collector_type[type(collector)],
INTERNALERROR>           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
INTERNALERROR> KeyError: <class '_pytest.doctest.DoctestModule'>
```

https://github.com/ipython/ipykernel/actions/runs/6921554050/job/18827290458?pr=1167

**Comment by seifertm** [comment]
@blink1073 Thanks for testing with pre-release versions. These kinds of bug reports are incredibly helpful!
The issue should be fixed as of v0.23.0b0.

**Comment by blink1073** [comment]
[Confirmed](https://github.com/ipython/ipykernel/actions/runs/6998882970), thank you!
